### PR TITLE
Dev/115 Delete Exam Modal

### PR DESCRIPTION
**Pull request by Chaoyuuu**
image

### CODE REVIEW

File: src/app/exam/options/ExamOptions.js [hunk 1]
```diff
@@ -44,61 +47,79 @@ const ExamOptions = () => {
...
+                                <section>
+                                    <button className="button update-button"
+                                            onClick={onButtonUpdateChangeClick}>
+                                        Update Change
+                                    </button>
+                                </section>
+                            </div>
+                            <div className="column right">
+                            </div>
+                            <TitleLine title={"Danger Zone"}/>
```
Comment (nyngwang) [inline src/app/exam/options/ExamOptions.js:100]: Remove the `{ }` pair.

File: src/app/exam/options/ExamOptions.js [hunk 2]
```diff
@@ -44,61 +47,79 @@ const ExamOptions = () => {
...
+                            <DangerZone onDangerButtonClick={() => setShowDeleteExamModal(true)}
+                                        header={'Delete this exam'}
```
Comment (nyngwang) [inline src/app/exam/options/ExamOptions.js:102]: ditto.

File: src/app/exam/options/ExamOptions.js [hunk 3]
```diff
@@ -44,61 +47,79 @@ const ExamOptions = () => {
...
+                                        description={'Once you delete an exam, there is no going back. Please be certain.'}
+                                        buttonName={'Delete Exam'}/>
```
Comment (nyngwang) [inline src/app/exam/options/ExamOptions.js:104]: ditto.

File: src/app/exam/options/ExamOptions.js (hunk 1 above)
Comment (nyngwang) [reply src/app/exam/options/ExamOptions.js:100]: Ditto.

File: src/app/exam/options/ExamOptions.js [hunk 4]
```diff
@@ -44,61 +47,79 @@ const ExamOptions = () => {
...
+                    currentURL={currentURL}
+                    examName={currentExam.name}
+                    examId={examId}/>
+                <div className="font-poppins" style={{paddingTop: "20px", paddingBottom: "150px"}}>
+                    <div style={{display: "flex", justifyContent: "flex-start"}}>
+                        <div style={{
+                            display: "flex", flexDirection: "column", alignItems: "flex-start",
+                            paddingLeft: "150px"
+                        }}>
+                            <TitleLine title={"Options"}/>
```
Comment (nyngwang) [inline src/app/exam/options/ExamOptions.js:75]: ditto.

File: src/app/exam/options/ExamOptions.js [hunk 5]
```diff
@@ -44,61 +47,79 @@ const ExamOptions = () => {
...
-                                <ExamWhiteList />
-                            </section>
-                            <section>
-                                <button className="button update-button"
-                                    onClick={onButtonUpdateChangeClick}>
-                                    Update Change
-                                </button>
-                            </section>
+        <>
+            {shouldRedirect ? <Redirect to={`/exams`}/> : ""}
```
Comment (nyngwang) [inline src/app/exam/options/ExamOptions.js:63]: Use `"/exams"`.

File: src/app/exam/options/ExamOptions.js [hunk 6]
```diff
@@ -44,61 +47,79 @@ const ExamOptions = () => {
...
-                        </div>
-                        <TitleLine title={"Danger Zone"} />
-                        <DangerZone onDangerButtonClick={() => deleteExam()}
-                            header={'Delete this exam'}
-                            description={'Once you delete an exam, there is no going back. Please be certain.'}
-                            buttonName={'Delete Exam'} />
                     </div>
                 </div>
+
+                <DeleteConfirmationModal title={"Delete the Exam?"}
```
Comment (nyngwang) [inline src/app/exam/options/ExamOptions.js:109]: ditto.

**Review by nyngwang — CHANGES_REQUESTED** [review]
Review Done.

**Comment by Wally5077** [comment]
Figma 上四周都是圓角，但你的似乎下方是直角
**Figma:**
image

**Your:**
image

File: src/app/exam/options/ExamOptions.js [hunk 7]
```diff
@@ -44,61 +47,79 @@ const ExamOptions = () => {
     };
 
     const deleteExam = () => {
-        console.log("delete exam")
+        examService.deleteExam(examId).then(() => {
```
Comment (Wally5077) [inline src/app/exam/options/ExamOptions.js:50]: 請把 .then 換行一下

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done.

If admin does delete the exam successfully, will the exam list still show deleted exam when back to the exam list?
if the answer is 'Yes', you may need to create new issue for this.
bty, It's just my friendly reminder. 👌

**Comment by Chaoyuuu** [comment]
> Figma 上四周都是圓角，但你的似乎下方是直角
> **Figma:**
> image
> 
> **Your:**
> image



> Review done.
> 
> If admin does delete the exam successfully, will the exam list still show deleted exam when back to the exam list?
> if the answer is 'Yes', you may need to create new issue for this.
> bty, It's just my friendly reminder. 👌


For the first comment about the UI modal, I have no idea why it looks different than other `DeleteConfirmationModal`. 😢  I have created an issue for this.

For the second comment, this function has already been supported in this issue :)

**Review by nyngwang — APPROVED** [review]
LGTM.

**Review by Wally5077 — APPROVED** [review]
LGTM

**Pull request merged**